src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/UriExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis.Razor
+{
+    internal static class UriExtensions
+    {
+        public static string GetAbsoluteOrUNCPath(this Uri uri)
```

Comment:
@ajaybhargavb  What if you want to support any other type of path in the future? You'd have to change the function name and everyplace it's called.

Reply:
Uri will take care of handling that and providing the correct AbsolutePath. UNC is the only special case 😄